daemon/cleanup.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/cilium/cilium/pkg/pidfile"
+)
+
+func handleInterrupt() <-chan struct{} {
+	// Handle the handleOSSignals
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGQUIT, syscall.SIGINT, syscall.SIGHUP, syscall.SIGTERM)
+	interrupt := make(chan struct{})
+	go func() {
+		for s := range sig {
```

Comment:
I'm pretty sure I've seen this pattern before, so maybe it's the canonical way to do this (the loop). We always break at the end of it, though, so it seems a little pointless? Relatedly, it might be good to keep this loop running and printing, as a way to know when many signals are being sent.

Reply:
It's basically going through all the signals and finding which one triggered, then logging which of the signals it actually was - so the break isn't exactly pointless. It just breaks on the first signal that was found (after logging which it was).

Reply:
Oh, I thought only triggered signals would sent on this channel by `signal.Notify`, unlike a select syscall. How does it check which one triggered in this case? I might need to audit any code I've written that uses `signal.Notify` :/